File: BurgurFrontend/src/app/Model/Cliente/cliente.ts
export interface Cliente {
    id: number;
    nombre: string;
    apellido: string;
    correo: string;
    contrasena?: string; // Opcional para no exponer en frontend
    telefono: string;
    direccion: string;
    fechaRegistro: Date;
    activo: boolean;
    pedidos?: number[]; // Solo IDs de pedidos
}

export interface ClienteRegistro {
    nombre: string;
    apellido: string;
    correo: string;
    contrasena: string;
    confirmPassword?: string;
    telefono: string;
    direccion: string;
}

export interface ClienteLogin {
    correo: string;
    contrasena: string;
}

File: burger-club_original/burgur/src/main/java/restaurante/example/burgur/Model/Cliente.java
//burger-club/burgur/src/main/java/restaurante/example/burgur/Model/Cliente.java
package restaurante.example.burgur.Model;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Cliente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String apellido;
    private String correo;
    private String contrasena;
    private String telefono;
    private String direccion;
    private boolean activo;

    //Relaciones BDD
    //1). Relación Pedido - Cliente (1 a muchos)
    @OneToMany(mappedBy = "cliente",
           cascade = {CascadeType.PERSIST, CascadeType.MERGE},
           orphanRemoval = false)
    private List<Pedido> pedidos = new ArrayList<>();

    

}